pkg/latte/intent_approve_login_link.go

```diff
@@ -0,0 +1,79 @@
+package latte
```

Comment:
1. This PR need not depend on https://github.com/authgear/authgear-server/pull/2912
2. IntentApproveLoginLink takes nothing to create.
3. IntentApproveLoginLink take an input which contains the code.
4. IntentApproveLoginLink ReactTo the input. It checks the code. If the code is correct, set SetUserInputtedMagicLinkCode and produce a new node NodeVerifiedLoginLinkCode.
5. NodeVerifiedLoginLinkCode is a deadend.
6. IntentApproveLoginLink becomes finished if it has 1 node (NodeVerifiedLoginLinkCode).